tc_core_analyzer_lib/tests/integration/test_inconsistently_active.py

```diff
@@ -0,0 +1,156 @@
+    UW_DEG_THR = 1  # minimum number of accounts interacted with to be active
+    PAUSED_T_THR = 1  # time period to remain paused
+    CON_T_THR = 4  # time period to be consistent active
+    CON_O_THR = 3  # time period to be consistent active
+    EDGE_STR_THR = 5  # minimum number of interactions for connected
+    UW_THR_DEG_THR = 5  # minimum number of accounts for connected
+    VITAL_T_THR = 4  # time period to assess for vital
+    VITAL_O_THR = 3  # times to be connected within VITAL_T_THR to be vital
+    STILL_T_THR = 2  # time period to assess for still active
+    STILL_O_THR = 2  # times to be active within STILL_T_THR to be still active
+    # time periods into the past (history) to be newly active for computing dropped
+    DROP_H_THR = 2
+    # consecutive time periods into the past to have not been active for computing
+    DROP_I_THR = 1
+
+    act_param = [
+        INT_THR,
+        UW_DEG_THR,
+        PAUSED_T_THR,
+        CON_T_THR,
+        CON_O_THR,
+        EDGE_STR_THR,
+        UW_THR_DEG_THR,
+        VITAL_T_THR,
+        VITAL_O_THR,
+        STILL_T_THR,
+        STILL_O_THR,
+        DROP_H_THR,
+        DROP_I_THR,
+    ]
+
+    int_mat = {
+        DiscordActivity.Reply: np.zeros((acc_count, acc_count)),
+        DiscordActivity.Mention: np.zeros((acc_count, acc_count)),
+        DiscordActivity.Reaction: np.zeros((acc_count, acc_count)),
+    }
+
+    # `user_1` intracting with `user_2`
+    int_mat[DiscordActivity.Reaction][0, 1] = 2
+
+    activities = [
+        DiscordActivity.Reaction,
+        DiscordActivity.Mention,
+        DiscordActivity.Reply,
+    ]
+
+    engagement = EngagementAssessment(
+        activities=activities, activities_ignore_0_axis=[], activities_ignore_1_axis=[]
+    )
+
+    # the analytics
+    for w_i in range(max_interval):
+        # time window
+        WINDOW_D = 7
+
+        (_, *activity_dict) = engagement.compute(
+            int_mat=int_mat,
+            w_i=w_i,
+            acc_names=acc_names,
+            act_param=act_param,
+            WINDOW_D=WINDOW_D,
+            **activity_dict,
+        )
+        if w_i == 14:
+            int_mat[DiscordActivity.Reaction][0, 1] = 0
+
+        activity_dict = dict(zip(memberactivities, activity_dict))
+
+    print("all_consistent:", activity_dict["all_consistent"])
+    print("all_paused:", activity_dict["all_paused"])
+    print("all_inconsistent:", activity_dict["all_inconsistent"])
+
+    assert activity_dict["all_inconsistent"] == {
+        "0": set(),
+        "1": set(),
+        "2": set(),
+        "3": set(),
+        "4": set(),
+        "5": set(),
+        "6": set(),
+        "7": {"user0", "user1"},
+        "8": {"user0", "user1"},
+        "9": {"user0", "user1"},
+        "10": {"user0", "user1"},
+        "11": {"user0", "user1"},
+        "12": {"user0", "user1"},
+        "13": {"user0", "user1"},
+        "14": set(),  # was incluedd in all_paused here
```

Comment:
Is this comment correct? If the activity stops on day 14 I would expect it to be all_paused until day 20. Furthermore, all_paused should not exclude it being all_inconsistent. As a result, I'm not sure why day 14 is not counted as all_inconsistent.

Reply:
I think I've added this comment wrongly when I was copying and pasting some parts of the other tests to this test. So the results for this test is as blow
```
all_consistent: {'0': set(), '1': set(), '2': set(), '3': set(), '4': set(), '5': set(), '6': set(), '7': set(), '8': set(), '9': set(), '10': set(), '11': set(), '12': set(), '13': set(), '14': {'user0', 'user1'}, '15': set(), '16': set(), '17': set(), '18': set(), '19': set(), '20': set(), '21': {'user0', 'user1'}, '22': set(), '23': set(), '24': set(), '25': set(), '26': set(), '27': set()}
all_paused: {'0': set(), '1': set(), '2': set(), '3': set(), '4': set(), '5': set(), '6': set(), '7': set(), '8': set(), '9': set(), '10': set(), '11': set(), '12': set(), '13': set(), '14': set(), '15': {'user0', 'user1'}, '16': {'user0', 'user1'}, '17': {'user0', 'user1'}, '18': {'user0', 'user1'}, '19': {'user0', 'user1'}, '20': {'user0', 'user1'}, '21': {'user0', 'user1'}, '22': set(), '23': set(), '24': set(), '25': set(), '26': set(), '27': set()}
all_inconsistent: {'0': set(), '1': set(), '2': set(), '3': set(), '4': set(), '5': set(), '6': set(), '7': {'user0', 'user1'}, '8': {'user0', 'user1'}, '9': {'user0', 'user1'}, '10': {'user0', 'user1'}, '11': {'user0', 'user1'}, '12': {'user0', 'user1'}, '13': {'user0', 'user1'}, '14': set(), '15': {'user0', 'user1'}, '16': {'user0', 'user1'}, '17': {'user0', 'user1'}, '18': {'user0', 'user1'}, '19': {'user0', 'user1'}, '20': {'user0', 'user1'}, '21': set(), '22': set(), '23': set(), '24': set(), '25': set(), '26': set(), '27': set()}
```